fastlane/lib/fastlane/actions/device_grid/device_grid.rb

```diff
@@ -0,0 +1,104 @@
+        def run(languages: nil, devices: nil, prefix_command: nil)
+          public_key = nil
+          [".fastlane/appetize_public_key.txt", "fastlane/appetize_public_key.txt"].each do |current| # not using Dir[...] since it doesn't show hidden files
+            public_key = File.read(current).strip if File.exist?(current)
+          end
+          raise("No public_key file found, make sure to run fastlane with `generate_device_grid` before calling `device_grid` in danger") if public_key.to_s.length == 0
+
+          devices ||= %w(iphone4s iphone5s iphone6s iphone6splus ipadair)
+          languages ||= ["en"]
+          prefix_command = "bundle exec" if File.exist?("Gemfile")
+          prefix_command ||= ""
+
+          deep_link_matches = pr_body.match(/:link:\s(.*)/) # :link: emoji
+          deep_link = deep_link_matches[1] if deep_link_matches
+
+          ENV["FASTLANE_DISABLE_COLORS"] = "true" # since we fetch the URL from the output
+
+          markdown("<table>")
+          languages.each do |current_language|
+            markdown("<tr>")
+              markdown("<td>")
+                markdown("<b>#{current_language[0..1]}</b>")
+              markdown("</td>")
+
+              devices.each do |current_device|
+                markdown("<td>")
+
+                params = {
+                  public_key: public_key,
+                  language: current_language,
+                  device: current_device
+                }
+                params[:launch_url] = deep_link if deep_link
+                params_str = params.collect { |k, v| "#{k}:\"#{v}\"" }.join(" ")
+                url = `#{prefix_command} fastlane run appetize_viewing_url_generator #{params_str}`
+                url = url.match(%r{Result:.*(https\:\/\/.*)})[1].strip
+
+                markdown("<a href='#{url}'>")
+                  markdown("<p align='center'>")
+                    markdown("<img height='130' src='#{url_for_device(current_device)}' />")
+                    markdown("<br />")
+                    markdown(beautiful_device_name(current_device))
+                  markdown("</p>")
+                markdown("</a>")
+
+                markdown("</td>")
+              end
+            markdown("</tr>")
+          end
+          markdown("</table>")
+        end
+
+        def beautiful_device_name(str)
+          return {
+            iphone4s: "iPhone 4s",
+            iphone5s: "iPhone 5s",
+            iphone6s: "iPhone 6s",
+            iphone6splus: "iPhone 6s Plus",
+            ipadair: "iPad Air",
+            iphone6: "iPhone 6",
+            iphone6plus: "iPhone 6 Plus",
+            ipadair2: "iPad Air 2",
+            nexus5: "Nexus 5",
+            nexus7: "Nexus 7",
+            nexus9: "Nexus 9"
+          }[str.to_sym] || str.to_s
+        end
+
+        def url_for_device(str)
+          str = str.to_sym
+          host = "https://raw.githubusercontent.com/fastlane/fastlane/master/fastlane/lib/fastlane/actions/device_grid/assets/"
```

Comment:
I have two suggestions for this:

1. Link to the tagged version from fastlane
```
fastlane_version `fastlane -v`
host = "https://raw.githubusercontent.com/fastlane/fastlane/#{fastlane_version}/..."
```

2. Move this functionality into `AppetizeViewingUrlGeneratorAction` and shell out, so we can control moving where these assets are hosted from later.